### PR TITLE
Converted codebase to TS

### PR DESCRIPTION
**Pull request by dshubhadeep**
### Semver Classification

- [ ] This PR only includes documentation or non-code changes.
- [ ] This PR fixes a bug and does not change the (intended) API/interface.
- [ ] This PR adds methods or properties to the API/interface.
- [x] This PR removes or renames methods or properties in the API/interface.

### CODE REVIEW

**Comment by kyranet** [comment]
Thanks you very much for this! The code looks good, though the code style isn't very similar to the previous, if you're able, I would like the new TypeScript lint rules to be derived from [eslint-config-marine](https://github.com/iCrawl/eslint-config-marine), removing `eslint-config-klasa`.

The `.eslintrc.json` should look like this (to keep most of the previous style):

```json
{
	"extends": "marine/node",
	"rules": {
		"@typescript-eslint/explicit-function-return-type": 0,
		"@typescript-eslint/no-use-before-define": ["error", { "typedefs": false, "functions": false }],
		"@typescript-eslint/promise-function-async": 0,
		"no-dupe-class-members": 0,
		"no-use-before-define": ["error", { "functions": false }],
		"padded-blocks": ["error", { "classes": "always" }],
		"operator-linebreak": ["error", "before"]
	}
}
```

And this would require `eslint@^5.16.0`, `eslint-config-marine@^2.2.2`, and `@typescript-eslint/eslint-plugin@^1.9.0`, installed as dev dependencies (for this to work).

Also, for the npm commands to work properly, the scripts have to be changed to `eslint --ext ts src test`, as by default, `eslint` ignores `.ts` files.

I think `eslint --ext ts src test --fix` should re-format everything for you, removing most of the warnings, I can fix the strict errors later, you can switch `"strict": false` in `tsconfig.json` if you want, and I'll take care of those errors, otherwise... for the nullable issue, you can assume they're not nullable by doing `object!.property` if I recall correctly, using the [non-null assertion operator](https://github.com/Microsoft/TypeScript/wiki/What's-new-in-TypeScript#non-null-assertion-operator).

File: src/lib/Node.ts [hunk 1]
```diff
@@ -0,0 +1,211 @@
+import { EventEmitter } from 'events';
+import NodeSocket from './Structures/NodeSocket';
+import NodeServer from './Structures/NodeServer';
+import { NodeOptions, SendOptions, BroadcastOptions } from 'veza';
+
+class Node extends EventEmitter {
+
+	/**
+   * @typedef {Object} NodeOptions
```
Comment (kyranet) [inline src/lib/Node.ts]: This comment and many others are pretty broken, *maybe someday ESLint will lint JSDoc's indentations*.
```suggestion
	 * @typedef {Object} NodeOptions
```

Comment (dshubhadeep) [reply src/lib/Node.ts]: Done. 👍

Comment (kyranet) [reply src/lib/Node.ts]: Not quite yet, there are still lines mixing tabs and spaces. In GitHub it's pretty easy to find them as the tabs take 8 spaces, and if you look at the diff, you can find them easily.

My tricks are to select everything in the very middle, if there's a tab it'll select either the start of the line or the end of the padding, and if there's a space, it'll select the middle of it. Kind of like this:

Comment (dshubhadeep) [reply src/lib/Node.ts]: Ok, I'll look into it. 
What do you want me use ? Tabs or spaces ?
Adding 2 tabs messed up the alignment, that's why I had to use tab+space.

Comment (kyranet) [reply src/lib/Node.ts]: I prefer tabs. Also, there's a tool in Visual Studio Code to convert all spaces to tabs, though I'm not sure if it also converts the padding from comment blocks.

- `Ctrl + Shift + P`.
- `> Convert indentation to Tabs`.

Comment (dshubhadeep) [reply src/lib/Node.ts]: Is this fine ?

Comment (kyranet) [reply src/lib/Node.ts]: There's some weird `\t\s\t` that should be `\t\t\s`, but that'd be nitpicking a lot. Do the tests still work after this rewrite? They're not run by the CI yet.

Comment (dshubhadeep) [reply src/lib/Node.ts]: No, the tests won't work since there were some issues I couldn't resolve properly.

Comment (kyranet) [reply src/lib/Node.ts]: Oh, ok, I'll merge this later and fix them, again, thanks you very much for this!

**Pull request merged**